src/DirectSolver.cpp

```diff
@@ -23,8 +23,8 @@ namespace blitzdg {
 
     void DirectSolver::solve(const real_matrix_type& A, const real_matrix_type& B, real_matrix_type& X) const {
 
-        firstIndex ii;
-        secondIndex jj;
+        if ( !(isRowMajor(A) == isRowMajor(B) ) )
+            throw runtime_error("Matrices A and B must have same storage order!");
 
```

Comment:
Since you copy A and B into pod arrays I am not sure this check is needed. The code should still work if they have different storage orders.

Reply:
nope, you were right. whoops.